Realm/RLMRealm.mm

```diff
@@ -395,7 +335,7 @@ + (instancetype)realmWithPath:(NSString *)path
     }
 
     // try to reuse existing realm first
-    RLMRealm *realm = cachedRealm(path);
+    RLMRealm *realm = RLMGetCurrentThreadCachedRealmForPath(path);
```

Comment:
This sounds a bit awkward. `RLMGetThreadLocalCachedRealmForPath` sounds a bit better. This isn't a big deal though.